src/wayland/shm/mod.rs

```diff
@@ -366,6 +366,147 @@ pub fn has_alpha(format: wl_shm::Format) -> bool {
     )
 }
 
+macro_rules! shm_format_table {
+    (
+        $(
+            $fourcc: ident => $shm: ident
+        ),* $(,)?
+    ) => {
+        /// Convert from a [`Fourcc`](backend::allocator::Fourcc) format to a wl_shm format
+        pub const fn fourcc_to_shm_format(value: $crate::backend::allocator::Fourcc) -> Option<$crate::reexports::wayland_server::protocol::wl_shm::Format> {
```

Comment:
Only if we had impl const...

Reply:
Even then this doesn't work, because of the orphan rule. `Fourcc` and `wl_shm` are defined in different crates.

Reply:
Well less so for a From impl and more so that we can just map Argb8888 and Xrgb8888 and the using wl_shm::Format::from(u32).

Reply:
yeah, right